### DISCUSSION TITLE
ESPHome component Ethernet-SPI W5500: connection problem

### DISCUSSION THREAD

**Discussion by AndySymons** [spali/esphome-components#5] [Q&A]
First, thanks for implementing this valuable addition to ESPHome. 

Second, I hope this is posted in the right place. The discussions link from the [Ethernet-SPI page](https://github.com/spali/esphome-components/tree/master/components/ethernet_spi) took me to the [M5Stack PoE Camera board configuration page](https://github.com/spali/esphome-components/discussions).

Now my problem: I am pairing an [Espressif Devkit v4](https://www.ebay.co.uk/itm/254803716289) with an [AZDelivery W5500 Ethernet module](https://www.amazon.co.uk/dp/B07CYVRMPL). The configuration yaml is compiled and loaded correctly, and the program initially says it is connected, but after that all requests are timed out. I do not know if this is a bug or whether I missed something? 

Config: 
```
ethernet:
  type: W5500
  cs_pin: 18
  clk_pin: 19
  mosi_pin: 21
  miso_pin: 23
  reset_pin: 22
  interrupt_pin: 36

  manual_ip:
    static_ip: 192.168.0.42
    gateway: 192.168.0.1
    subnet: 255.255.255.0

```
Notes:
- [ ] I connected RST and INT as well as the four SPI pins. I get GPIO numbering errors if I do not connect these, so they do not appear to be 'optional' as the documentation says
- [ ] I left the clock speed as default. I get an error if I try to set it to 80MHz. The message says that the software will select the best speed anyway, while the documentation says it is 29.97MHz. 
- [ ] I am using a manual IP because when I did not, I got a DNS error message saying it could not resolve the address, though the documentation says manual IP is optional 
- [ ] After loading, I tried to access the API from a browser at 192.168.0.42:6053. This said ‘unreachable’. I also tried the OTA port 192.168.0.42:3232. This one timed out.
- [ ] I then tried to load the program again over the network. This gave a timed out connection. 

By 'documentation' I mean[ this page](https://esphome.io/components/ethernet.html). (I did not find anything with more detail)
 
Log file from program load: 
[installation_log.txt](https://github.com/spali/esphome-components/files/15228949/installation_log.txt)

Log file when I tried to update the firmware over the network connection:
[otn_load_log.txt](https://github.com/spali/esphome-components/files/15228970/otn_load_log.txt)

I need advice on how to debug this!

**Comment by spali** [comment]
You may mix my old "hacky" W5500 component here with the official one that exists since march.
You should ignore everything in this repo regarding W5500 and only check the official documentation online and maybe you find some answers also in the PR of the official solution:
https://github.com/esphome/esphome/pull/4424

**Reply by AndySymons** [reply]
Ah yes, that is confusing. I do want the official one. How do I tell the difference?

**Reply by spali** [reply]
official repository or web page:
https://github.com/esphome/esphome
https://esphome.io/components/ethernet.html

**Comment by AndySymons** [comment]
I did go through that before posting. I do not understand all the technical detail, but much of it is historical, so presumably in the release, or applies to different host boards from the one I am using. 

I notice for example that some of the examples have  phy_addr:, but in the release that is not accepted for W5500, so is presumably set automatically. 

The fact that it compiles,loads and initially seems to connect suggests that my configuration is complete that it is not an SPI problem of talking to the interface. It seems to be an incomplete Ethernet setup such that it fails to respond to incoming requests. The trouble is, I have no idea how to debug the problem. I hoped maybe an expert would spot some clue in the log files that I cannot see, or suggest some test that might reveal the problem.

**Reply by spali** [reply]
I'm not really an expert... especially with SPI. And also not of the esphome network stack. You may find better answers if you ask in the esphome community for debugging help.
